datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -602,15 +622,38 @@ impl BufferedBatch {
             + mem::size_of::<Range<usize>>()
             + mem::size_of::<usize>();
 
+        let num_rows = batch.num_rows();
         BufferedBatch {
             batch,
             range,
             join_arrays,
             null_joined: vec![],
             size_estimation,
             join_filter_failed_idxs: HashSet::new(),
+            num_rows,
+            spill_file: None,
         }
     }
+
+    fn spill_to_disk(
+        &mut self,
+        path: RefCountedTempFile,
+        buffered_schema: SchemaRef,
+        batch_size: usize,
+    ) -> Result<()> {
+        let batch = std::mem::replace(
+            &mut self.batch,
+            RecordBatch::new_empty(Arc::clone(&buffered_schema)),
+        );
+        let _ = spill_record_batch_by_size(
```

Comment:
Should this return value be ignored? It seems like we should pass the error

Reply:
good catch, fixed